src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -57,9 +62,20 @@ public void writeColumnHeaders(final List<String> columnNames) throws IOExceptio
   public void writeTuples(final ReadableTable tuples) throws IOException {
     final String[] row = new String[tuples.numColumns()];
     /* Serialize every row into the output stream. */
+    Schema tbsc = tuples.getSchema();
     for (int i = 0; i < tuples.numTuples(); ++i) {
       for (int j = 0; j < tuples.numColumns(); ++j) {
-        row[j] = tuples.getObject(j, i).toString();
+        Type type = tbsc.getColumnType(j);
+        if (type.equals(Type.BLOB_TYPE)) {
+          // write the file out
+          // add filename to the csv file
+          String filename = UUID.randomUUID().toString();
```

Comment:
They all go in the same directory with no prefix, if the UUID exists, it gets overwritten.  I will use getTempFile for now, but there really isn't a benefit for our use case here. Using simpler implementation is easier as the more mature APIs often bring dependency and complexity that might not be needed.